Tools/TestWebKitAPI/Tests/WebCore/ImageBufferTests.cpp

```diff
@@ -282,6 +285,41 @@ TEST_P(PreserveResolutionOperationTest, SinkIntoImageWorks)
     EXPECT_TRUE(hasTestPattern(*verifyBuffer));
 }
 
+// ImageBuffer test fixture for tests that are variant to the image buffer device scale factor and options
+class AnyScaleTest : public testing::TestWithParam<std::tuple<float, TestImageBufferOptions>> {
+public:
+    float deviceScaleFactor() const { return std::get<0>(GetParam()); }
+    OptionSet<ImageBufferOptions> imageBufferOptions() const
+    {
+        auto testOptions = std::get<1>(GetParam());
+        if (testOptions == TestImageBufferOptions::Accelerated)
+            return ImageBufferOptions::Accelerated;
+        return { };
+    }
+};
+
+// Test that ImageBuffer::getPixelBuffer() returns PixelBuffer that is sized to the ImageBuffer::logicalSize() * ImageBuffer::resolutionScale().
+TEST_P(AnyScaleTest, GetPixelBufferDimensionsContainScale)
+{
+    IntSize testSize { 50, 57 };
+    auto buffer = ImageBuffer::create(testSize, RenderingPurpose::Unspecified, deviceScaleFactor(), DestinationColorSpace::SRGB(), PixelFormat::BGRA8, imageBufferOptions());
```

Comment:
Does not this create a local ImageBuffer? Should not we create a RemoteImageBuffer for this change?

Reply:
The commit message explains this: this is expressing the API contract and invoking it with the WebCore side code. We don't yet test the WebKit side code.